asterius/rts/rts.memory.mjs

```diff
@@ -4,17 +4,58 @@ function mask(n) {
   return (BigInt(1) << BigInt(n)) - BigInt(1);
 }
 
+/**
+ * Class acting as the low-level interface to Wasm memory.
+ * It mainly provides methods to load/store data in memory
+ * (e.g. {@link Memory#i64Load}, {@link Memory#i64Store}),
+ * static methods to handle pointer tagging (e.g. {@link Memory#getTag},
+ * {@link Memory#getDynTag}), and a MBlock allocator
+ * ({@link Memory#getMBlocks} and {@link Memory#freeMBlocks}).
+ */
 export class Memory {
   constructor() {
+    /**
+     * The underlying Wasm Memory instance.
+     * @name Memory#memory
+     */
     this.memory = undefined;
+    /**
+     * The number of MBlock slots reserved for
+     * the static part of memory (vs the dynamic part
+     * where heap objects are allocated at runtime).
+     * The static MBlocks contain the initial compiled
+     * Wasm code plus auxiliary static data structures
+     * like info tables.
+     * @name Memory#staticMBlocks
+     */
     this.staticMBlocks = undefined;
+    /**
+     * Low-level interfaces for reading/writing the contents
+     * of {@link Memory#memory}.
+     * @name Memory#i8view
+     * @name Memory#dataView
+     */
     this.i8View = undefined;
     this.dataView = undefined;
+    /**
```

Comment:
Maybe we should explain further what "capacity" means here. We separate between "size" and "capacity", and when we need to grow the underlying wasm linear memory, we always grow it at least by a factor of 2, to reduce amortized overhead of allocating individual mblocks; this is a well known technique of implementing things like `std::vector` in C++. So "capacity" is the real size of wasm linear memory, and size indicates the range within the capacity that we really use at the moment.